### DISCUSSION TITLE
Running statements from a file?

### DISCUSSION THREAD

**Discussion by saward** [rusqlite/rusqlite#1117] [Questions] [answered]
Hi,

I'm wondering if there are recommend ways to have a bunch of statements in files, to be run by rusqlite?  E.g., having a default database defined by a bunch of statements in a file, or having a migration from one version to another?

Thanks!

**Comment by gwenn** [comment]
Not strictly related to `rusqlite`.
Check first if there is no tool to help managing migration.
Or maybe https://sqlite.org/pragma.html#pragma_user_version + migration scripts.
```
if pragma('user_version') <= 0 {
  // init schema
  pragma('user_version', 1);
}
if pragma('user_version') <= 1 {
  // alter schema
  pragma('user_version', 2);
}
...
```

**Comment by thomcc** [comment, marked as answer]
You can also do `conn.execute_batch(include_str!("my_file.sql"))` and such.

**Comment by saward** [comment]
Great, thanks, exactly what I was after!  @gwenn that looks useful too, thanks for the tip.
